tests/test_mimegroups.py

```diff
@@ -0,0 +1,46 @@
+import pytest
+
+from xtractmime import mimegroups
+
+
+ALL_MIME_GROUPS = {
+    name[3:-10]: getattr(mimegroups, name)
+    for name in dir(mimegroups)
+    if name.startswith("is_") and name.endswith("_mime_type")
+}
+
+
+@pytest.mark.parametrize(
+    "mime_type,mime_groups",
+    [
+        (b"image/foo", {"image"}),
+        (b"image/x-icon", {"image"}),
+        (b"audio/foo", {"audio_video"}),
+        (b"audio/basic", {"audio_video"}),
+        (b"video/foo", {"audio_video"}),
+        (b"video/avi", {"audio_video"}),
+        (b"application/ogg", {"audio_video"}),
+        (b"font/foo", {"font"}),
+        (b"application/vnd.ms-fontobject", {"font"}),
+        (b"application/font-cff", {"font"}),
+        (b"application/foo+zip", {"zip"}),
+        (b"application/zip", {"zip", "archive"}),
+        (b"application/x-gzip", {"archive"}),
+        (b"application/foo+xml", {"xml", "scriptable"}),
+        (b"text/xml", {"xml", "scriptable"}),
+        (b"text/html", {"html", "scriptable"}),
+        (b"application/pdf", {"scriptable"}),
+        (b"application/ecmascript", {"javascript"}),
+        (b"application/foo+json", {"json"}),
+        (b"text/json", {"json"}),
+        (b"application/postscript", {"text"}),
+        (b"text/plain", {}),
+        (b"text/plain; charset=ISO-8859-1", {}),
+        (b"application/octet-stream", {}),
```

Comment:
To cover all MIME type essences mentioned in the standard, we should also include:

- From https://mimesniff.spec.whatwg.org/#mime-type-groups:
  - All font MIME type essences
  - `application/x-rar-compressed`
  - `application/xml`
  - All JavaScript MIME type essences
- All image MIME type essences from https://mimesniff.spec.whatwg.org/#matching-an-image-type-pattern
- All audio or video MIME type essences from https://mimesniff.spec.whatwg.org/#matching-an-audio-or-video-type-pattern (not just those on the table, see the steps afterwards)
- All font MIME type essences from https://mimesniff.spec.whatwg.org/#matching-a-font-type-pattern
- The 3 unknown MIME type essences from step 1 of https://mimesniff.spec.whatwg.org/#determining-the-computed-mime-type-of-a-resource
- From https://mimesniff.spec.whatwg.org/#sniffing-a-mislabeled-feed
  - `application/rss+xml`
  - `application/atom+xml`
- `text/vtt` from https://mimesniff.spec.whatwg.org/#sniffing-in-a-text-track-context
- `text/cache-manifest` from https://mimesniff.spec.whatwg.org/#sniffing-in-a-cache-manifest-context

Also, this list is bound to grow, so please keep these tuples sorted, either by where in the standard they appear preceded by a comment about it as I suggested originally, or alphabetically.